router/src/main/resources/interceptor/lib/utils.lua

```diff
@@ -86,3 +90,43 @@ function table.shallow_copy(src, des)
         des[k] = v
     end
 end
+
+--- log body and headers retrieved from the handle
+---@param handle table
+---@param log_message string
+function wire_log_body(handle, log_message, wire_log_enabled)
+    if wire_log_enabled then 
+        local log_output = "\n" .. "[wirelog]" .. log_message .. "\n"
+        if handle:body() then
+            handle:logInfo(log_output .. handle:body():getBytes(0, handle:body():length()) .. log_output)
+        else 
+            handle:logInfo("Body is empty")
+        end
+    end
+end
+
+--- log body and headers retrieved from the handle
+---@param handle table
+---@param log_message string
+function wire_log_headers(handle, log_message, wire_log_enabled)
+    if wire_log_enabled then 
+        local headers = handle:headers()
+        local log_output = "\n"
+        for header_name, header_value in pairs(headers) do
+            log_output = log_output .. "[wirelog]" .. log_message .. header_name .. ": " .. header_value .. "\n"
+        end
+        handle:logInfo(log_output)
+    end
+end
+
+--- log body and headers retrieved from the handle
+---@param log_message_body string
+---@param log_message_header string
+function utils.wire_log_body_and_headers(handle, log_message_body, log_message_header, wire_log_enabled)
+    if wire_log_enabled then 
+        wire_log_body(handle, log_message_body, wire_log_enabled)
+        wire_log_headers(handle, log_message_header, wire_log_enabled)
```

Comment:
We need to log trailers too. https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter

Reply:
added trailer logging as well